scylla/src/transport/session.rs

```diff
@@ -1666,7 +1666,10 @@ impl Session {
             .consistency_set_on_statement
             .unwrap_or(execution_profile.consistency);
 
+        let mut query_plan_is_empty = true;
```

Comment:
Wanted to do a peekable check to ensure the iterator is not empty like:
```rust
query_plan.peekable()
```

but it caused this lifetime error:
```
error: higher-ranked lifetime error
   --> scylla/src/transport/session_test.rs:409:5
    |
409 | /     tokio::spawn(async move {
410 | |         let values = (
411 | |             (1_i32, 2_i32, "abc"),
412 | |             (),
...   |
415 | |         session_clone.batch(&batch, values).await.unwrap();
416 | |     })
    | |______^
    |
    = note: could not prove `[async block@scylla/src/transport/session_test.rs:409:18: 416:6]: std::marker::Send`
    ```
    So, I just defaulted to doing this more simple check